### PR TITLE
fix: correctly detect the active Node js version during headless installation

### PR DESCRIPTION
**Pull request by dy-dx**
This is a proof of concept for fixing https://github.com/pnpm/pnpm/issues/5266.

When pnpm is bundled with node.js (via standalone installer), and `engine-strict=true`, pnpm falls back to using the bundled node.js.

pnpm properly detects the system node.js version when installing new packages, but not when doing headless installs (i.e. when the lockfile is up to date).

I think the issue is the `extendOptions` function:
https://github.com/pnpm/pnpm/blob/e71ca7579e5d2a9142595a9fe2ab70b826dad802/pkg-manager/core/src/install/index.ts#L234

It sets `options.nodeVersion = process.version`, i.e. whatever node is bundled with pnpm.

This doesn't seem to matter during the code path for non-headless installations.
But during headless installations, it prevents pnpm from checking the system Node.js version when checking package compatibility.

### CODE REVIEW

**Comment by zkochan** [comment]
But the same code is present here: https://github.com/pnpm/pnpm/blob/e71ca7579e5d2a9142595a9fe2ab70b826dad802/config/package-is-installable/src/index.ts#L89

So does this change actually make any difference?

File: pkg-manager/core/src/install/extendInstallOptions.ts [hunk 1]
```diff
@@ -187,7 +188,7 @@ const defaults = (opts: InstallOptions) => {
     },
     lockfileDir: opts.lockfileDir ?? opts.dir ?? process.cwd(),
     lockfileOnly: false,
-    nodeVersion: process.version,
+    nodeVersion: opts.nodeVersion ?? getSystemNodeVersion(),
```
Comment (zkochan) [inline pkg-manager/core/src/install/extendInstallOptions.ts]: This should be enough
```suggestion
    nodeVersion: opts.nodeVersion,
```

Comment (zkochan) [reply pkg-manager/core/src/install/extendInstallOptions.ts]: The issue is that we pass `process.version`. `getSystemNodeVersion` will be used anyway inside `package-is-installable`.

**Review by zkochan — APPROVED** [review]

**Comment by dy-dx** [comment]
> So you just need to replace `process.version` with `opts.nodeVersion`. No need to run `getSystemNodeVersion`

@zkochan yes I agree, I actually tried that first and I can confirm that it also solves the issue. But I wasn't sure if the resulting typing changes were going to be problematic or not.

Thanks for the quick response!

**Pull request merged**